components/playlist/renderer/playlist_render_frame_observer.cc

```diff
@@ -0,0 +1,68 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/playlist/renderer/playlist_render_frame_observer.h"
+
+#include "content/public/renderer/render_frame.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+#include "third_party/blink/public/web/web_script_source.h"
+
+namespace playlist {
+
+PlaylistRenderFrameObserver::PlaylistRenderFrameObserver(
+    content::RenderFrame* render_frame)
+    : RenderFrameObserver(render_frame),
+      RenderFrameObserverTracker<PlaylistRenderFrameObserver>(render_frame) {}
+
+PlaylistRenderFrameObserver::~PlaylistRenderFrameObserver() {}
+
+void PlaylistRenderFrameObserver::RunScriptsAtDocumentStart() {
+  if (!url_.SchemeIsHTTPOrHTTPS())
+    return;
+
+  // TODO(sko) This list should be dynamically updated from browser process.
+  // For now, we hardcode the list of domains that we want to run scripts at.
+  if (url_.host() == "www.youtube.com") {
```

Comment:
looks great!

but pls consider this instead of keeping `url_`:
```
static const base::NoDestructor<std::vector<blink::WebSecurityOrigin>> origins{
    {{blink::WebSecurityOrigin::Create(GURL("https://www.youtube.com"))}}};

render_frame()->GetWebFrame()->GetSecurityOrigin()->IsSameOriginWith(...);
```

Reply:
Cool! Done.